File: src/api/expiryRisk.ts
import apiClient from './client';

export interface ExpiryRiskItem {
  inventory_id: string;
  lot_id: string;
  sku_id: string;
  sku_name: string;
  category: string;
  location_id: string;
  location_name: string;
  available_qty: number;
  on_hand_qty: number;
  manufacturing_date: string;
  expiry_date: string;
  analysis_date: string;
  days_to_expiry: number;
  recent_30d_sales_qty: number;
  recent_average_daily_demand: number;
  forecast_consumption_before_expiry: number;
  forecast_method: string;
  projected_surplus: number;
  projected_shortage: number;
  surplus_ratio: number;
  urgency_factor: number;
  risk_score: number;
  risk_band: 'Expired' | 'Critical' | 'High' | 'Medium' | 'Low';
  explanation: string;
  proposed_actions: string[];
  fefo_position: number;
  related_recommendation_ids: string[];
  pack_size?: string | null;
  public_product_id?: string | null;
  product_name?: string | null;
  source_url?: string | null;
}

export interface ExpiryRiskSummaryCounts {
  expired_count: number;
  critical_count: number;
  high_count: number;
  medium_count: number;
  low_count: number;
  total_projected_surplus: number;
}

export interface ExpiryRiskListResponse {
  items: ExpiryRiskItem[];
  total: number;
  summary: ExpiryRiskSummaryCounts;
}

export interface ExpiryRiskFilterParams {
  risk_band?: string;
  sku_id?: string;
  category?: string;
  location_id?: string;
  expiry_bucket?: string;
  search?: string;
}

export async function getExpiryRiskList(params?: ExpiryRiskFilterParams): Promise<ExpiryRiskListResponse> {
  const res = await apiClient.get<ExpiryRiskListResponse>('/api/expiry-risk', { params });
  return res.data;
}

export async function getExpiryRiskDetail(lotId: string): Promise<ExpiryRiskItem> {
  const res = await apiClient.get<ExpiryRiskItem>(`/api/expiry-risk/${lotId}`);
  return res.data;
}

File: backend/app/schemas/expiry_risk.py
"""Pydantic schemas for Expiry Risk endpoints."""
from typing import List, Optional
from pydantic import BaseModel


class ExpiryRiskItem(BaseModel):
    inventory_id: str
    lot_id: str
    sku_id: str
    sku_name: str
    category: str
    location_id: str
    location_name: str
    available_qty: int
    on_hand_qty: int
    manufacturing_date: str
    expiry_date: str
    analysis_date: str
    days_to_expiry: int
    recent_30d_sales_qty: int
    recent_average_daily_demand: float
    forecast_consumption_before_expiry: float
    forecast_method: str
    projected_surplus: float
    projected_shortage: float
    surplus_ratio: float
    urgency_factor: float
    risk_score: float
    risk_band: str
    explanation: str
    proposed_actions: List[str]
    fefo_position: int
    related_recommendation_ids: List[str]
    pack_size: Optional[str] = None
    public_product_id: Optional[str] = None
    product_name: Optional[str] = None
    source_url: Optional[str] = None


class ExpiryRiskSummaryCounts(BaseModel):
    expired_count: int
    critical_count: int
    high_count: int
    medium_count: int
    low_count: int
    total_projected_surplus: float


class ExpiryRiskListResponse(BaseModel):
    items: List[ExpiryRiskItem]
    total: int
    summary: ExpiryRiskSummaryCounts
